ui/app/components/auth-jwt.js

```diff
@@ -171,6 +171,14 @@ export default Component.extend({
       if (e && e.preventDefault) {
         e.preventDefault();
       }
+      try {
+        await this.fetchRole.perform(this.roleName, { debounce: false });
+      } catch (error) {
+        // this task could be cancelled if the instances in didReceiveAttrs resolve after this was started
+        if (error?.name !== 'TaskCancelation') {
+          throw error;
+        }
+      }
```

Comment:
I can't see an issue with moving this above the conditional. If anything, since a role is required it should have come before in the first place to ensure we have the necessary data.

Reply:
That's my understanding? I'm guessing it wasn't above before because firing it off in `didReceiveAttrs` seems like it'd be sufficient. Unfortunately the conditionals that wrap `fetchRole` in that function don't account for the namespace updating